OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs

```diff
@@ -200,4 +206,172 @@ void RemoveBeacon(Beacon beacon)
+		readonly SupportPowerManager manager;
+		readonly string cursor;
+		readonly Animation targetCursor;
+		readonly string[] arrows = { "arrow-r", "arrow-br", "arrow-b", "arrow-bl", "arrow-l", "arrow-tl", "arrow-t", "arrow-tr" };
+		readonly Arrow[] directionArrows;
+
+		CPos targetCell;
+		int2 location;
+		bool startedDragging;
+		int facingBetween;
+
+		public SelectDirection(World world, string order, SupportPowerManager manager, string cursor)
+		{
+			this.order = order;
+			this.manager = manager;
+			this.cursor = cursor;
+
+			targetCursor = new Animation(world, "airstrike");
+			targetCursor.PlayRepeating("cursor");
+
+			directionArrows = GetArrows(world, 512, arrows.Length);
+		}
+
+		public IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			if (mi.Button == MouseButton.Right)
+			{
+				world.CancelInputMode();
+				yield break;
+			}
+
+			if (mi.Button == MouseButton.Left && mi.Event == MouseInputEvent.Down)
+			{
+				targetCell = cell;
+				location = mi.Location;
+				startedDragging = true;
+				yield break;
+			}
+
+			if (mi.Button == MouseButton.Left && mi.Event == MouseInputEvent.Up)
+			{
+				var ret = OrderInner().FirstOrDefault();
+				if (ret == null)
+					yield break;
+
+				world.CancelInputMode();
+				yield return ret;
+			}
+		}
+
+		IEnumerable<Order> OrderInner()
+		{
+			yield return new Order(order, manager.Self, Target.FromCell(manager.Self.World, targetCell), false)
+			{
+				SuppressVisualFeedback = true,
+				ExtraData = IsOutsideDragZone ? (uint)facingBetween : uint.MaxValue
+			};
+		}
+
+		public void Tick(World world)
+		{
+			targetCursor.Tick();
+
+			// Cancel the OG if we can't use the power
+			if (!manager.Powers.ContainsKey(order))
+				world.CancelInputMode();
+		}
+
+		bool IsOutsideDragZone
+		{
+			get { return (Viewport.LastMousePos - location).Length > 20; }
+		}
+
+		public IEnumerable<IRenderable> Render(WorldRenderer wr, World world)
+		{
+			yield break;
+		}
+
+		public IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world)
+		{
+			if (!startedDragging)
+				return Enumerable.Empty<IRenderable>();
+
+			var mousePos = Viewport.LastMousePos;
+			var palette = wr.Palette("chrome");
+			var worldPos = wr.Viewport.ViewToWorldPx(location);
+			var pos = wr.ProjectedPosition(worldPos);
+			var renderables = new List<IRenderable>(targetCursor.Render(pos, WVec.Zero, -511, palette, 1f));
+
+			if (IsOutsideDragZone)
+			{
+				var d = AngleOf(location, mousePos);
+				var s = GetArrow(d);
+				var towards = wr.Viewport.ViewToWorld(location);
+				var cPos = towards + s.Direction;
+				facingBetween = world.Map.FacingBetween(towards, cPos, 0);
+				renderables.Add(new SpriteRenderable(s.Sprite, pos, s.Offset, -511, palette, 1f, true));
+			}
+
+			return renderables;
+		}
+
+		public string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			return startedDragging ? "invisible" : cursor;
```

Comment:
This seems to be returning "invisible" slightly too early - I see a flash when clicking where neither real or fake cursor are visible.

Reply:
Could it be that this is caused by that we have two animations playing which are not in sync?